Hardware/Hardware/CardReader/StripeCardReader/StripeCardReaderService.swift

```diff
@@ -207,6 +207,29 @@ extension StripeCardReaderService: CardReaderService {
         }
     }
 
+    public func waitForInsertedCardToBeRemoved() -> Future<Void, Error> {
```

Comment:
nit: Since it never fails we could have the declaration as

`public func waitForInsertedCardToBeRemoved() -> Future<Void, Never>`

That would be also meaningful for the clients of `CardReaderService`. But now I see that in `NoOpCardReaderService` it always fails right?

Reply:
> nit: Since it never fails we could have the declaration as
> 
> `public func waitForInsertedCardToBeRemoved() -> Future<Void, Never>`
> 
> That would be also meaningful for the clients of `CardReaderService`. But now I see that in `NoOpCardReaderService` it always fails right?

Great suggestion 👍 updated in https://github.com/woocommerce/woocommerce-ios/pull/6665/commits/f0b4a6f5e5c1845d840be7444a21dd483f433442. Since `waitForInsertedCardToBeRemoved` is the last step of the card reader connection and payment flow, it would not be executed if any of the previous steps fails with an error. Also, `NoOpCardReaderService` is only used in Mac Catalyst (which doesn't support payments) and is not officially available yet